src/freertos_drivers/common/FlashFile.hxx

```diff
@@ -82,7 +82,7 @@ public:
     /// Overrides behavior of open for O_TRUNC.
     int open(File *file, const char *path, int flags, int mode) override
     {
-        if ((mode & O_WRONLY) && (flags & O_TRUNC))
+        if (flags & O_TRUNC)
```

Comment:
This is not correct. It should only be truncated if O_WRONLY or O_RDWR. If O_RDONLY, O_TRUNC should be ignored. The code should look like this:
`if ((flags & O_TRUNC) && (flags & O_ACCMODE) != O_RDONLY)`

From the standard "The result of using O_TRUNC without either O_RDWR or O_WRONLY is undefined." Though I do not think we should be truncating a file opened for read only in any case.

Reply:
done